src/server/middleware/sendHtml.js

```diff
@@ -57,7 +57,27 @@ export function safeSend(res, ...payload) {
   }
 }
 
-export function renderStaticErrorPage(res) {
+let errorPage;
+
+export async function setErrorPage(fallbackUrl) {
+  const response = await fetch(fallbackUrl);
+  const contentType = response.headers.get('content-type');
+  const contentLength = response.headers.get('content-length');
+
+  // If the Content-Type is not text/html throw an error
+  if (!contentType.includes('text/html')) {
+    throw new Error('Content-Type was not of type text/html');
+  }
+  // If the content length is over 50kb throw an error
+  if (contentLength > 50000) {
+    throw new Error('Content-Length was over 50Kb');
+  }
+  // Read the response as text.
+  errorPage = await response.text();
+  return errorPage;
```

Comment:
What happens when this function throws? will the root module fail to load?

We are setting the value of errorPage and then returning, is that a side effect here? @JAdshead any memory implications?

Reply:
The server is killed like @mtomcal and the rest of us discussed

Reply:
If the server is undergoing a fresh startup and an invalid Root Module is provided, then the server will fail to start up. This is the same behavior as trying to start  up a Root Module with a bad CSP configuration which results in a server exit today. See the previous comment scenario about a stable running server loading an invalid Root Module: https://github.com/americanexpress/one-app/pull/281#discussion_r481297652

Reply:
For reference: After some discussions with the team we decided that killing the server on an invalid custom error page is not ideal. Unlike the CSP the custom error page functionality is not critical for the server to work correctly since we can always fallback to the static error page as before. furthermore the custom error page is loaded from an external source which might not be reliable at some point when the server is trying to restart (due to autoscaling or pod restart)